### DISCUSSION TITLE
Is AdonisJS 5 Production Ready?

### DISCUSSION THREAD

**Discussion by RomainLanz** [adonisjs/core#1257] [General]
🤔

**Comment by RomainLanz** [comment]
Yes.

Read this: https://github.com/adonisjs/core/discussions/1257#discussioncomment-40511

**Reply by popovicieduard** [reply]
Making sure another question like: https://www.reddit.com/r/node/comments/7x3afi/anyone_using_adonisjs_in_production/ won't be asked ?

**Reply by dajpes** [reply]
Is there a way to migrate an existing adonis v4.1 project to adonis v5, or is it too late?

**Reply by pk4all** [reply]
I think no way because Adonis  V5 use typescript.

**Reply by pishguy** [reply]
@pk4all then we must waiting to release stable version of framework and we should know typeScript, is it right?

**Reply by pk4all** [reply]
@MahdiPishguy  Stable version is ready to use but yes we should know basics of typeScript.

**Reply by malpaso** [reply]
Really confusing having Adonis5 released yet the heaps of packages that make Adonis compelling to use are holding back adoption. Why is the conversion of those packages not a priority? Surely if Adonis5 was ready to be released that would be the focus now?

**Reply by DaveLomber** [reply]
If I do not like TS - can I still migrate to 5 ?

**Reply by McSneaky** [reply]
All JS == perfectly valid TS

So all you do is write regular JS in .ts files (except model property decorators, where you have to use @column() etc)

**Reply by DaveLomber** [reply]
But, if I do not want to use TS at all? Is there a way to continue use JS with Adonis 5?

**Comment by hariprasath-yadav** [comment]
i am confused with adonis js releases
if v5 is production-ready why the docs still remain in the "preview" domain
and i can't find the production release date of v5
anyone, please clarify my doubts

**Reply by RomainLanz** [reply]
Hey @hariprasath-yadav! 👋 

As stated in the AdonisJS 5 introduction article and in AdonisJS 5 documentation, we are using the name `Preview` because all modules from AdonisJS 4 have not been migrated yet. _For example, if your application relies on Social Authentication it would not be possible to migrate to AdonisJS 5 at the time._

![image](https://user-images.githubusercontent.com/2793951/88146611-6383ca80-cbfc-11ea-948b-e8d08f97ccbc.png)

If you want to learn more about what we plan for AdonisJS 5 and is currently "missing", feel free to check our [Roadmap](https://trello.com/b/3klaHbfP).

Also, ensure to properly read [our documentation](https://preview.adonisjs.com/guides/quick-start) and what we wrote on [our social media](https://twitter.com/adonisframework), we communicate a lot and every question you may have are already been answered.

![image](https://user-images.githubusercontent.com/2793951/88146549-523abe00-cbfc-11ea-8bbf-290ebb3c08e4.png)

**Comment by kilobyte2007** [comment]
What is the status now? I am starting a new project on Adonis v4 and thought I might as well start with v5, so my questions would be:
1. Are all the packages v5 ready?
2. Will it be possible to migrate a v4 project to v5, and if yes, will there be a migration guide?

Thanks a lot for this framework, love the v4 version, v5 seems to be even cooler.

**Reply by thetutlage** [reply]
Go with v5. We are already preparing for the final release and only a few things changes between `preview` and the upcoming `main` release

**Reply by kilobyte2007** [reply]
Thank you! I'll try to do that as I already have something on v4 :D
The only thing missing will probably be the Persona package I think, any updates on that one?

**Reply by meideros** [reply]
Will version 4  be unmaintained in the future?

**Comment by meideros** [comment]
Will version 4 still be maintained and usable?

**Reply by RomainLanz** [reply]
It will not be heavily maintained but there will be a small updates in the following days to fix few bugs and upgrade dependencies.

We recommend everyone to switch to V5 once its released.

More information available here: https://github.com/adonisjs/core/discussions/2416

**Reply by meideros** [reply]
ok I hope the web socket fonctionnality will be inside the v5 release !
